## Entendimento dos Dados

Este notebook será utilizado para analisar um recorte de dados referente ao ano de 2025. Antes de fazer toda a carga histórica, será importado apenas dados de 2025 para analisar a estrutura do dataset, quais são as colunas disponíveis, o que cada uma delas está representando e se realmente está de acordo com o que precisamos para o desenvolvimento deste projeto.

### 1. Importação das Bibliotecas

In [1]:
from pathlib import Path
import pandas as pd

DATA_PATH = Path('../data/raw')

### 2. Carregando e Compreendendo os Dados

In [2]:
electric_load_2025 = pd.read_parquet(DATA_PATH / 'curva_carga_2025.parquet')
electric_load_2025.head()

,id_subsistema,nom_subsistema,din_instante,val_cargaenergiahomwmed
0,N,NORTE,2025-01-01 00:00:00,7097.751
1,NE,NORDESTE,2025-01-01 00:00:00,12251.182
2,S,SUL,2025-01-01 00:00:00,10554.603
3,SE,SUDESTE,2025-01-01 00:00:00,35374.033
4,N,NORTE,2025-01-01 01:00:00,7053.337


#### 2.1. Dicionário do Dataset

Cada linha está representando uma carga elétrica que foi registrada em um subsistema durante uma determinada hora. Então, por exemplo, no dia 1 de Janeiro de 2025, à meia-noite, o subsistema NORDESTE registrou uma carga de 7097.75 MWmed, isso quer dizer que, naquele horário, o sistema precisou atender uma demanda média de aproximadamente 7 mil megawatts para aquela determinada região.

Abaixo está cada coluna existente e o que cada uma está trazendo de informação:

| Coluna | Descrição |
|---|---|
| id_subsistema | Sigla do subsistema elétrico |
| nom_subsistema | Nome do subsistema |
| din_instante | Data e hora de referência |
| val_cargaenergiahomwmed | Carga elétrica média no período, medida em MWmed |

#### 2.2. Verificando a Dimensão

In [3]:
print(f'O dataset possui um total de {electric_load_2025.shape[0]} linhas e {electric_load_2025.shape[1]} colunas.')

O dataset possui um total de 35040 linhas e 4 colunas.


#### 2.3. Informações Gerais

In [4]:
electric_load_2025.info(memory_usage='deep')


<class 'pandas.DataFrame'>
RangeIndex: 35040 entries, 0 to 35039
Data columns (total 4 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   id_subsistema            35040 non-null  str           
 1   nom_subsistema           35040 non-null  str           
 2   din_instante             35040 non-null  datetime64[ns]
 3   val_cargaenergiahomwmed  35040 non-null  float64       
dtypes: datetime64[ns](1), float64(1), str(2)
memory usage: 1.3 MB


#### 2.4. Confirmação de Anos e Horários

Nesta etapa, será verificado o ano inicial (menor ano) e o ano final (maior ano), para entender se o recorte de dados que fizemos está representando corretamento o intervalo que selecionamos. Além disso, também será observado a quantidade de horários distintos que temos.

In [5]:
print(f"Data inicial: {electric_load_2025['din_instante'].min()}")
print(f"Data final: {electric_load_2025['din_instante'].max()}")
print(f"Horários distintos: {electric_load_2025['din_instante'].nunique():,}")

Data inicial: 2025-01-01 00:00:00
Data final: 2025-12-31 23:00:00
Horários distintos: 8,760


#### 2.5. Verificando Valores Únicos dos Subsistemas

Agora, iremos observar quais são os subsistemas existentes em nosso conjunto de dados, eliminando qualquer tipo de duplicidade e mantendo apenas os registros que aparecem pelo menos uma vez nesse recorte de dados.

In [6]:
subsystems = electric_load_2025[['id_subsistema', 'nom_subsistema']].drop_duplicates().sort_values('id_subsistema')

subsystems


,id_subsistema,nom_subsistema
0,N,NORTE
1,NE,NORDESTE
2,S,SUL
3,SE,SUDESTE


#### 2.6. Frequência Absoluta dos Subsistemas

Abaixo, teremos a frequência que cada subsistema aparece no conjunto de dados.

In [7]:
electric_load_2025['nom_subsistema'].value_counts()

nom_subsistema
NORTE       8760
NORDESTE    8760
SUL         8760
SUDESTE     8760
Name: count, dtype: int64

### Conclusão Final do Notebook

O arquivo de 2025 contém registros entre 1 de janeiro e 31 de dezembro, com 8.760 horários distintos. Além disso, os quatro subsistemas possuem a mesma quantidade de registros, representando cerca de 25% do dataset total cada um.

A verificação completa de duplicidades, horários ausentes e consistência dos valores será realizada após a coleta de todo o histórico.